src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -192,17 +192,12 @@ static Object* pysec_cell(Section* sec) {
 
 static int NpySObj_contains(PyObject* s, PyObject* obj, const char* string) {
     /* Checks is provided PyObject* s contains obj */
-    PyObject* obj_seg;
-    int result;
     if (!PyObject_HasAttrString(obj, string)) {
         return 0;
     }
-    Py_INCREF(obj);
-    obj_seg = PyObject_GetAttrString(obj, string);
-    Py_DECREF(obj);
-    result = PyObject_RichCompareBool(s, obj_seg, Py_EQ);
-    Py_XDECREF(obj_seg);
-    return (result);
+    auto _pyobj = nb::borrow(obj);  // keep refcount+1 during use
```

Comment:
does `obj` need to have it's refcount preserved?

Reply:
Good observation. Indeed the object should have its ref count preserved at > 0 for the whole duration of the function. Nevertheless, without being absolutely sure, I feel inclined to make functional changes on a later PR, and leave this one as a pure refactoring

Reply:
Yes! Create an issue and merge this without changes. Then revisit it soon. (Same for the other comment.)

Reply:
Thank you for noticing!